packages/samples/headless-react/src/components/triggers/redirection-trigger.class.tsx

```diff
@@ -0,0 +1,45 @@
+import {Component, ContextType} from 'react';
+import {
+  buildRedirectionTrigger,
+  RedirectionTrigger as HeadlessRedirectionTrigger,
+  RedirectionTriggerState,
+  Unsubscribe,
+} from '@coveo/headless';
+import {AppContext} from '../../context/engine';
+
+export class RedirectionTrigger extends Component<{}, RedirectionTriggerState> {
+  static contextType = AppContext;
+  context!: ContextType<typeof AppContext>;
+
+  private controller!: HeadlessRedirectionTrigger;
+  private unsubscribe: Unsubscribe = () => {};
+
+  componentDidMount() {
+    this.controller = buildRedirectionTrigger(this.context.engine!);
+    this.updateState();
+    this.unsubscribe = this.controller.subscribe(() => this.updateState());
+  }
+
+  componentWillUnmount() {
+    this.unsubscribe();
+  }
+
+  componentDidUpdate() {
```

Comment:
This function is not really needed, since the redirection can happen inside the `updateState` function.
The `setState` call is not necessary because the `render` function is not visual.

I would rename `updateState` to `redirect` and add the `window.location.href` line there,